src/test/run-pass/init-large-type.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Makes sure that zero-initializing large types is reasonably fast,
+// Doing it incorrectly causes massive slowdown in LLVM during
+// optimisation.
+
+#![feature(intrinsics)]
+
+extern "rust-intrinsic" {
+    pub fn init<T>() -> T;
+}
+
+const SIZE: usize = 512 * 1024 * 1024;
```

Comment:
1024 * 1024 is enough to reliably reproduce the issue. No need to make this test require 512MiB of RAM to run?